src/magic_folder/snapshot.py

```diff
@@ -210,6 +324,124 @@ def deserialize_dict(snapshot_dict, author):
 
         return deserialize_dict(local_snapshot_dict, author)
 
+@attr.s
+class RemoteSnapshot(object):
+    """
+    Represents a snapshot corresponding to a particular version of a
+    file authored by a particular human.
+
+    :ivar name: the name of this Snapshot. This is a mangled path
+        relative to our local magic-folder path.
+
+    :ivar metadata: a dict containing metadata about this Snapshot.
+
+    :ivar parents_raw: list of capablitiy-strings instances of our
+        parents
+
+    :ivar author: SnapshotAuthor instance
+
+    :ivar capability: an immutable CHK:DIR2 capability-string.
+    """
+
+    name = attr.ib()
+    author = attr.ib()  # any SnapshotAuthor instance
+    metadata = attr.ib()
+    capability = attr.ib()
+    parents_raw = attr.ib()
+    content_cap = attr.ib()
```

Comment:
So, this is a thing I *don't* like about attrs: there's no private attributes. So "normal users of this code" shouldn't use this, they should use `fetch_content` .. but in any case, I will document :)

Reply:
What kind of privacy do you want?  Does `_content_cap = attr.ib()` do something pretty good here?  attrs will still add it to `__init__` but it will drop the leading `_` there so `RemoteSnapshot(content_cap=...)` is still fine but `RemoteSnapshot(...).content_cap` will be an `AttributeError` (`RemoteSnapshot(...)._content_cap` still works, of course).